packages/clay-drop-down/src/Menu.tsx

```diff
@@ -70,30 +70,54 @@ const ClayDropDownMenu = React.forwardRef<HTMLDivElement, IProps>((
 		hasLeftSymbols,
 		hasRightSymbols,
 		onSetActive,
+		style,
 		...otherProps
 	}: IProps,
 	// TS + refs don't always play nicely together, which is why it is casted
 	// in so many places below.
 	// See https://github.com/microsoft/TypeScript/issues/30748#issuecomment-480197036
 	ref
 ) => {
+	const [region, setRegion] = React.useState<{
+		left: number;
+		top: number;
+	} | null>(null);
 	const subPortalRef = useRef<HTMLDivElement | null>(null);
 
 	useDropdownCloseInteractions([alignElementRef, subPortalRef], onSetActive);
 
 	useLayoutEffect(() => {
 		if (
+			active &&
 			alignElementRef.current &&
 			(ref as React.RefObject<HTMLDivElement>).current
 		) {
-			Align.align(
-				(ref as React.RefObject<HTMLElement>).current!,
-				alignElementRef.current,
-				alignmentPosition,
-				autoBestAlign
-			);
+			let newRegion: ReturnType<typeof Align.getAlignRegion>;
+
+			if (autoBestAlign) {
+				newRegion = Align.suggestAlignBestRegion(
+					(ref as React.RefObject<HTMLElement>).current!,
+					alignElementRef.current,
+					alignmentPosition
+				).region;
+			} else {
+				newRegion = Align.getAlignRegion(
+					(ref as React.RefObject<HTMLElement>).current!,
+					alignElementRef.current,
+					alignmentPosition
+				);
+			}
+
+			setRegion(newRegion);
```

Comment:
It seems that this does not solve some things, take a look at the "List Page" demo and try to open most DropDown, some are randomly misaligned.

Maybe this part of align code would solve this: https://github.com/metal/metal-plugins/blob/master/packages/metal-position/src/Align.js#L42-L56. I'm not sure, I'm just making an assumption.

Reply:
So I was debugging this and I think the ListPage demo is a particular edge case for the `autoBestAlign` feature from metal-position. If I go back to using the normal Align.align, it functions the same way. I think it has to do with the unique position it is next the the edge of the window. So I think this particular change should still be fine.

One other option is to entirely replace the metal-position library and use something that is more widely adopted. Maybe something like, https://www.npmjs.com/package/dom-align

What do you think?

Reply:
Actually I take that back... I was starting to see some other issues resulting from that change.

I decided to just remove that refactor and instead only fix the alignment issue.

Reply:
Hmm, I didn't realize it was already broken in ListPage (Seems to work fine here https://storybook.clayui.com/?path=/story/demos-templates--list-page). 

Since @jbalsas wants to depreciate metal-position, it may be time to test `dom-align`.